packages/code-snippet-block/src/settings.ts

```diff
@@ -0,0 +1,326 @@
+                    ],
+                },
+            ],
+            off: [
+                {
+                    id: 'padding',
+                    type: 'slider',
+                    defaultValue: '0px',
+                    choices: [
+                        {
+                            value: '0px',
+                            label: 'None',
+                        },
+                        {
+                            value: '6rem',
+                            label: 'S',
+                        },
+                        {
+                            value: '9rem',
+                            label: 'M',
+                        },
+                        {
+                            value: '15rem',
+                            label: 'L',
+                        },
+                    ],
+                },
+            ],
+        },
+    ],
+    style: [
+        {
+            id: 'theme',
+            type: 'dropdown',
+            defaultValue: DEFAULT_THEME_VALUE,
+            label: 'Color schema',
+            size: DropdownSize.Small,
+            choices: [
+                {
+                    value: 'default',
+                    label: 'Default Theme',
+                },
+                {
+                    value: 'dracula',
+                    label: 'Dracula',
+                },
+                {
+                    value: 'evaLight',
+                    label: 'Eva Light',
+                },
+                {
+                    value: 'gitHubDark',
+                    label: 'GitHub Dark',
+                },
+                {
+                    value: 'gitHubLight',
+                    label: 'GitHub Light',
+                },
+                {
+                    value: 'materialDarker',
+                    label: 'Material Darker',
+                },
+                {
+                    value: 'materialLighter',
+                    label: 'Material Lighter',
+                },
+                {
+                    value: 'materialPalenight',
+                    label: 'Material Palenight',
+                },
+                {
+                    value: 'oneDark',
+                    label: 'One Dark',
+                },
+                {
+                    value: 'oneLight',
+                    label: 'One Light',
+                },
+                {
+                    value: 'base16Dark',
+                    label: 'Base16 Dark',
+                },
+                {
+                    value: 'base16Light',
+                    label: 'Base16 Light',
+                },
+                {
+                    value: 'cobalt',
+                    label: 'Cobalt',
+                },
+            ],
+        },
+        {
+            id: 'withBorder',
+            type: 'switch',
+            defaultValue: true,
+            label: 'Border',
+        },
+        {
+            id: 'border',
+            type: 'multiInput',
```

Comment:
I would add `show: (bundle: Bundle) => bundle.getBlock('withBorder')?.value === true,`.

This way the user can only change settings related to border, when he/she wants selects border = true